### PR TITLE
Pioneer 16 Ingest CSVs

### PR DESCRIPTION
**Pull request by cdobs**
CP01CNSM - D00014
CP02PMCI - D00015
CP02PMCO - D00015
CP02PMUI - D00016
CP02PMUO - D00017
CP03ISSM - D00013
CP04OSPM - D00014
CP04OSSM - D00013
CP05MOAS-PG564 D0003
CP05MOAS-PG583 D0004

### CODE REVIEW

File: CP01CNSM/CP01CNSM_D00014_ingest.csv [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+mi.dataset.driver.cg_dcl_eng.dcl.cg_dcl_eng_dcl_telemetered_driver,/omc_data/whoi/OMC/CP01CNSM/D00014/cg_data/dcl12/syslog/*syslog.log,CP01CNSM-SBD12-00-DCLENG000,telemetered,Available,
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_telemetered_driver,/omc_data/whoi/OMC/CP01CNSM/D00014/cg_data/dcl12/hyd2/*hyd2.log,CP01CNSM-SBD12-03-HYDGN0000,telemetered,Available,
+mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_telemetered_driver,/omc_data/whoi/OMC/CP01CNSM/D00014/cg_data/dcl12/pco2a/*pco2a.log,CP01CNSM-SBD12-04-PCO2AA000,telemetered,Available,
+mi.dataset.driver.wavss_a.dcl.wavss_a_dcl_telemetered_driver,/omc_data/whoi/OMC/CP01CNSM/D00014/cg_data/dcl12/wavss/*wavss.log,CP01CNSM-SBD12-05-WAVSSA000,telemetered,Available,
+mi.dataset.driver.metbk_a.dcl.metbk_a_dcl_telemetered_driver,/omc_data/whoi/OMC/CP01CNSM/D00014/cg_data/dcl12/metbk2/*metbk2.log,CP01CNSM-SBD12-06-METBKA000,telemetered,Available,
+mi.dataset.driver.fdchp_a.dcl.fdchp_a_dcl_telemetered_driver,/omc_data/whoi/OMC/CP01CNSM/D00014/cg_data/dcl12/fdchp/*fdchp.log,CP01CNSM-SBD12-08-FDCHPA000,telemetered,Available,
```
Comment (snwhite) [inline CP01CNSM/CP01CNSM_D00014_ingest.csv:35]: Looks good.

File: CP03ISSM/CP03ISSM_D00013_ingest.csv [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#mi.dataset.driver.cg_stc_eng.stc.mopak_o_dcl_telemetered_driver,/omc_data/whoi/OMC/CP03ISSM/D00013/cg_data/dcl11/mopak/*.mopak.log,CP03ISSM-SBD11-01-MOPAK0000,telemetered,Not Expected,Data will be available upon recovery
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_telemetered_driver,/omc_data/whoi/OMC/CP03ISSM/D00013/cg_data/dcl11/hyd1/*hyd1.log,CP03ISSM-SBD11-02-HYDGN0000,telemetered,Available,
+mi.dataset.driver.metbk_a.dcl.metbk_a_dcl_telemetered_driver,/omc_data/whoi/OMC/CP03ISSM/D00013/cg_data/dcl11/metbk/*.metbk*.log,CP03ISSM-SBD11-06-METBKA000,telemetered,Available,
+mi.dataset.driver.cg_dcl_eng.dcl.cg_dcl_eng_dcl_telemetered_driver,/omc_data/whoi/OMC/CP03ISSM/D00013/cg_data/dcl12/syslog/*.syslog.log,CP03ISSM-SBD12-00-DCLENG000,telemetered,Available,
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_telemetered_driver,/omc_data/whoi/OMC/CP03ISSM/D00013/cg_data/dcl12/hyd2/*hyd2.log,CP03ISSM-SBD12-03-HYDGN0000,telemetered,Available,
+mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_telemetered_driver,/omc_data/whoi/OMC/CP03ISSM/D00013/cg_data/dcl12/pco2a/*.pco2a.log,CP03ISSM-SBD12-04-PCO2AA000,telemetered,Available,
```
Comment (snwhite) [inline CP03ISSM/CP03ISSM_D00013_ingest.csv:32]: Looks good.

File: CP04OSSM/CP04OSSM_D00013_ingest.csv [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+mi.dataset.driver.metbk_a.dcl.metbk_a_dcl_telemetered_driver,/omc_data/whoi/OMC/CP04OSSM/D00013/cg_data/dcl11/metbk/*.metbk.log,CP04OSSM-SBD11-06-METBKA000,telemetered,Available,
+mi.dataset.driver.cg_dcl_eng.dcl.cg_dcl_eng_dcl_telemetered_driver,/omc_data/whoi/OMC/CP04OSSM/D00013/cg_data/dcl12/syslog/*.syslog.log,CP04OSSM-SBD12-00-DCLENG000,telemetered,Available,
+mi.dataset.driver.hyd_o.dcl.hyd_o_dcl_telemetered_driver,/omc_data/whoi/OMC/CP04OSSM/D00013/cg_data/dcl12/hyd*/*hyd*.log,CP04OSSM-SBD12-03-HYDGN0000,telemetered,Available,
+mi.dataset.driver.pco2a_a.sample.pco2a_a_sample_telemetered_driver,/omc_data/whoi/OMC/CP04OSSM/D00013/cg_data/dcl12/pco2a/*.pco2a.log,CP04OSSM-SBD12-04-PCO2AA000,telemetered,Available,
+mi.dataset.driver.fdchp_a.dcl.fdchp_a_dcl_telemetered_driver,/omc_data/whoi/OMC/CP04OSSM/D00013/cg_data/dcl11/fdchp2/*fdchp2.log,CP04OSSM-SBD11-08-FDCHPB000,telemetered,Available,
+mi.dataset.driver.fdchp_a.dcl.fdchp_a_dcl_telemetered_driver,/omc_data/whoi/OMC/CP04OSSM/D00013/cg_data/dcl12/fdchp1/*fdchp1.log,CP04OSSM-SBD12-08-FDCHPA000,telemetered,Available,
```
Comment (snwhite) [inline CP04OSSM/CP04OSSM_D00013_ingest.csv:34]: Looks good.

File: CP02PMCI/CP02PMCI_D00015_ingest.csv [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+mi.dataset.driver.WFP_ENG.STC_IMODEM.wfp_eng_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMCI/D00015/imm/mmp/E*.DAT,CP02PMCI-WFP01-00-WFPENG000,telemetered,Available,
+mi.dataset.driver.vel3d_k.wfp_stc.vel3d_k_wfp_stc_telemetered_driver,/omc_data/whoi/OMC/CP02PMCI/D00015/imm/mmp/A*.DEC,CP02PMCI-WFP01-01-VEL3DK000,telemetered,Available,
+mi.dataset.driver.dofst_k.wfp.coastal_dofst_k_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMCI/D00015/imm/mmp/C*.DAT,CP02PMCI-WFP01-02-DOFSTK000,telemetered,Available,
+mi.dataset.driver.ctdpf_ckl.wfp.coastal_ctdpf_ckl_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMCI/D00015/imm/mmp/C*.DAT,CP02PMCI-WFP01-03-CTDPFK000,telemetered,Available,
+mi.dataset.driver.flort_kn.stc_imodem.flort_kn__stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMCI/D00015/imm/mmp/E*.DAT,CP02PMCI-WFP01-04-FLORTK000,telemetered,Available,
+mi.dataset.driver.PARAD_K.STC_IMODEM.parad_k_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMCI/D00015/imm/mmp/E*.DAT,CP02PMCI-WFP01-05-PARADK000,telemetered,Available,
```
Comment (snwhite) [inline CP02PMCI/CP02PMCI_D00015_ingest.csv:10]: Looks good.

File: CP02PMCO/CP02PMCO_D00015_ingest.csv [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+mi.dataset.driver.WFP_ENG.STC_IMODEM.wfp_eng_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMCO/D00015/imm/mmp/E*.DAT,CP02PMCO-WFP01-00-WFPENG000,telemetered,Available,
+mi.dataset.driver.vel3d_k.wfp_stc.vel3d_k_wfp_stc_telemetered_driver,/omc_data/whoi/OMC/CP02PMCO/D00015/imm/mmp/A*.DEC,CP02PMCO-WFP01-01-VEL3DK000,telemetered,Available,
+mi.dataset.driver.dofst_k.wfp.coastal_dofst_k_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMCO/D00015/imm/mmp/C*.DAT,CP02PMCO-WFP01-02-DOFSTK000,telemetered,Available,
+mi.dataset.driver.ctdpf_ckl.wfp.coastal_ctdpf_ckl_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMCO/D00015/imm/mmp/C*.DAT,CP02PMCO-WFP01-03-CTDPFK000,telemetered,Available,
+mi.dataset.driver.flort_kn.stc_imodem.flort_kn__stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMCO/D00015/imm/mmp/E*.DAT,CP02PMCO-WFP01-04-FLORTK000,telemetered,Available,
+mi.dataset.driver.PARAD_K.STC_IMODEM.parad_k_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMCO/D00015/imm/mmp/E*.DAT,CP02PMCO-WFP01-05-PARADK000,telemetered,Available,
```
Comment (snwhite) [inline CP02PMCO/CP02PMCO_D00015_ingest.csv:10]: Looks good.

File: CP02PMUI/CP02PMUI_D00016_ingest.csv [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+mi.dataset.driver.WFP_ENG.STC_IMODEM.wfp_eng_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMUI/D00016/imm/mmp/E*.DAT,CP02PMUI-WFP01-00-WFPENG000,telemetered,Available,
+mi.dataset.driver.vel3d_k.wfp_stc.vel3d_k_wfp_stc_telemetered_driver,/omc_data/whoi/OMC/CP02PMUI/D00016/imm/mmp/A*.DEC,CP02PMUI-WFP01-01-VEL3DK000,telemetered,Available,
+mi.dataset.driver.dofst_k.wfp.coastal_dofst_k_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMUI/D00016/imm/mmp/C*.DAT,CP02PMUI-WFP01-02-DOFSTK000,telemetered,Available,
+mi.dataset.driver.ctdpf_ckl.wfp.coastal_ctdpf_ckl_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMUI/D00016/imm/mmp/C*.DAT,CP02PMUI-WFP01-03-CTDPFK000,telemetered,Available,
+mi.dataset.driver.flort_kn.stc_imodem.flort_kn__stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMUI/D00016/imm/mmp/E*.DAT,CP02PMUI-WFP01-04-FLORTK000,telemetered,Available,
+mi.dataset.driver.PARAD_K.STC_IMODEM.parad_k_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMUI/D00016/imm/mmp/E*.DAT,CP02PMUI-WFP01-05-PARADK000,telemetered,Available,
```
Comment (snwhite) [inline CP02PMUI/CP02PMUI_D00016_ingest.csv:10]: Looks good.

File: CP02PMUO/CP02PMUO_D00017_ingest.csv [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+mi.dataset.driver.WFP_ENG.STC_IMODEM.wfp_eng_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMUO/D00017/imm/mmp/E*.DAT,CP02PMUO-WFP01-00-WFPENG000,telemetered,Available,
+mi.dataset.driver.vel3d_k.wfp_stc.vel3d_k_wfp_stc_telemetered_driver,/omc_data/whoi/OMC/CP02PMUO/D00017/imm/mmp/A*.DEC,CP02PMUO-WFP01-01-VEL3DK000,telemetered,Available,
+mi.dataset.driver.dofst_k.wfp.coastal_dofst_k_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMUO/D00017/imm/mmp/C*.DAT,CP02PMUO-WFP01-02-DOFSTK000,telemetered,Available,
+mi.dataset.driver.ctdpf_ckl.wfp.coastal_ctdpf_ckl_wfp_telemetered_driver,/omc_data/whoi/OMC/CP02PMUO/D00017/imm/mmp/C*.DAT,CP02PMUO-WFP01-03-CTDPFK000,telemetered,Available,
+mi.dataset.driver.flort_kn.stc_imodem.flort_kn__stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMUO/D00017/imm/mmp/E*.DAT,CP02PMUO-WFP01-04-FLORTK000,telemetered,Available,
+mi.dataset.driver.PARAD_K.STC_IMODEM.parad_k_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP02PMUO/D00017/imm/mmp/E*.DAT,CP02PMUO-WFP01-05-PARADK000,telemetered,Available,
```
Comment (snwhite) [inline CP02PMUO/CP02PMUO_D00017_ingest.csv:10]: Looks good.

File: CP04OSPM/CP04OSPM_D00014_ingest.csv [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+mi.dataset.driver.WFP_ENG.STC_IMODEM.wfp_eng_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP04OSPM/D00014/imm/mmp/E*.DAT,CP04OSPM-WFP01-00-WFPENG000,telemetered,Available,
+mi.dataset.driver.vel3d_k.wfp_stc.vel3d_k_wfp_stc_telemetered_driver,/omc_data/whoi/OMC/CP04OSPM/D00014/imm/mmp/A*.DEC,CP04OSPM-WFP01-01-VEL3DK000,telemetered,Available,
+mi.dataset.driver.dofst_k.wfp.coastal_dofst_k_wfp_telemetered_driver,/omc_data/whoi/OMC/CP04OSPM/D00014/imm/mmp/C*.DAT,CP04OSPM-WFP01-02-DOFSTK000,telemetered,Available,
+mi.dataset.driver.ctdpf_ckl.wfp.coastal_ctdpf_ckl_wfp_telemetered_driver,/omc_data/whoi/OMC/CP04OSPM/D00014/imm/mmp/C*.DAT,CP04OSPM-WFP01-03-CTDPFK000,telemetered,Available,
+mi.dataset.driver.flort_kn.stc_imodem.flort_kn__stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP04OSPM/D00014/imm/mmp/E*.DAT,CP04OSPM-WFP01-04-FLORTK000,telemetered,Available,
+mi.dataset.driver.PARAD_K.STC_IMODEM.parad_k_stc_imodem_telemetered_driver,/omc_data/whoi/OMC/CP04OSPM/D00014/imm/mmp/E*.DAT,CP04OSPM-WFP01-05-PARADK000,telemetered,Available,
```
Comment (snwhite) [inline CP04OSPM/CP04OSPM_D00014_ingest.csv:10]: Looks good.

File: CP05MOAS-PG564/CP05MOAS-PG564_D00003_ingest.csv [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+mi.dataset.driver.moas.gl.ctdgv.ctdgv_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG564/D00003/merged-from-glider/cp_*.mrg,CP05MOAS-PG564-01-CTDGVM000,telemetered,Available,
+mi.dataset.driver.moas.gl.dosta.dosta_abcdjm_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG564/D00003/merged-from-glider/cp_*.mrg,CP05MOAS-PG564-02-DOSTAM000,telemetered,Available,
+mi.dataset.driver.moas.gl.flort_m.flort_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG564/D00003/merged-from-glider/cp_*.mrg,CP05MOAS-PG564-03-FLORTM000,telemetered,Available,
+mi.dataset.driver.moas.gl.flort_o.flort_o_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG564/D00003/merged-from-glider/cp_*.mrg,CP05MOAS-PG564-04-FLORTO000,telemetered,Available,
+mi.dataset.driver.nutnr_m.glider.nutnr_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG564/D00003/merged-from-glider/cp_*.mrg,CP05MOAS-PG564-05-NUTNRM000,telemetered,Available,
+mi.dataset.driver.moas.gl.parad.parad_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG564/D00003/merged-from-glider/cp_*.mrg,CP05MOAS-PG564-06-PARADM000,telemetered,Available,
```
Comment (snwhite) [inline CP05MOAS-PG564/CP05MOAS-PG564_D00003_ingest.csv:8]: Looks good.

File: CP05MOAS-PG583/CP05MOAS-PG583_D00004_ingest.csv [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+mi.dataset.driver.moas.gl.ctdgv.ctdgv_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG583/D00004/merged-from-glider/cp_*.mrg,CP05MOAS-PG583-01-CTDGVM000,telemetered,Available,
+mi.dataset.driver.moas.gl.dosta.dosta_abcdjm_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG583/D00004/merged-from-glider/cp_*.mrg,CP05MOAS-PG583-02-DOSTAM000,telemetered,Available,
+mi.dataset.driver.moas.gl.flort_m.flort_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG583/D00004/merged-from-glider/cp_*.mrg,CP05MOAS-PG583-03-FLORTM000,telemetered,Available,
+mi.dataset.driver.moas.gl.flort_o.flort_o_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG583/D00004/merged-from-glider/cp_*.mrg,CP05MOAS-PG583-04-FLORTO000,telemetered,Available,
+mi.dataset.driver.nutnr_m.glider.nutnr_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG583/D00004/merged-from-glider/cp_*.mrg,CP05MOAS-PG583-05-NUTNRM000,telemetered,Available,
+mi.dataset.driver.moas.gl.parad.parad_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-PG583/D00004/merged-from-glider/cp_*.mrg,CP05MOAS-PG583-06-PARADM000,telemetered,Available,
```
Comment (snwhite) [inline CP05MOAS-PG583/CP05MOAS-PG583_D00004_ingest.csv:8]: Looks good.

**Review by snwhite — APPROVED** [review]
Ingest CSVs for Pioneer 16 telemetered data look good.

File: CP02PMUO/CP02PMUO_D00017_ingest.csv (hunk 7 above)
Comment (dbogorff) [reply CP02PMUO/CP02PMUO_D00017_ingest.csv:10]: Looks good!

File: CP05MOAS-GL339/CP05MOAS-GL339_D00013_ingest.csv [hunk 11]
```diff
@@ -0,0 +1,6 @@
+parser,filename_mask,reference_designator,data_source,status,notes
+mi.dataset.driver.moas.gl.engineering.glider_eng_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-GL339/D00013/merged-from-glider/cp_*.mrg,CP05MOAS-GL339-00-ENG000000,telemetered,Available,
+mi.dataset.driver.moas.gl.flort_m.flort_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-GL339/D00013/merged-from-glider/cp_*.mrg,CP05MOAS-GL339-02-FLORTM000,telemetered,Available,
+mi.dataset.driver.moas.gl.ctdgv.ctdgv_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-GL339/D00013/merged-from-glider/cp_*.mrg,CP05MOAS-GL339-03-CTDGVM000,telemetered,Available,
+mi.dataset.driver.moas.gl.dosta.dosta_abcdjm_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-GL339/D00013/merged-from-glider/cp_*.mrg,CP05MOAS-GL339-04-DOSTAM000,telemetered,Available,
+mi.dataset.driver.moas.gl.parad.parad_m_glider_telemetered_driver,/omc_data/whoi/OMC/CP05MOAS-GL339/D00013/merged-from-glider/cp_*.mrg,CP05MOAS-GL339-05-PARADM000,telemetered,Available,
```
Comment (snwhite) [inline CP05MOAS-GL339/CP05MOAS-GL339_D00013_ingest.csv:6]: Looks good.

**Review by snwhite — APPROVED** [review]

File: CP02PMUI/CP02PMUI_D00016_ingest.csv (hunk 6 above)
Comment (dbogorff) [reply CP02PMUI/CP02PMUI_D00016_ingest.csv:10]: Looks good!

File: CP02PMCO/CP02PMCO_D00015_ingest.csv (hunk 5 above)
Comment (dbogorff) [reply CP02PMCO/CP02PMCO_D00015_ingest.csv:10]: Looks good!

File: CP02PMCI/CP02PMCI_D00015_ingest.csv (hunk 4 above)
Comment (dbogorff) [reply CP02PMCI/CP02PMCI_D00015_ingest.csv:10]: Looks good!

**Review by dbogorff — APPROVED** [review]

File: CP03ISSM/CP03ISSM_D00013_ingest.csv (hunk 2 above)
Comment (dbogorff) [reply CP03ISSM/CP03ISSM_D00013_ingest.csv:32]: Looks good!

File: CP04OSSM/CP04OSSM_D00013_ingest.csv (hunk 3 above)
Comment (dbogorff) [reply CP04OSSM/CP04OSSM_D00013_ingest.csv:34]: Looks good!

**Pull request merged**